libr/anal/fcn.c

```diff
@@ -1862,12 +1862,21 @@ R_API RAnalFunction *r_anal_fcn_find_name(RAnal *anal, const char *name) {
 	return NULL;
 }
 
+#define MAXBBSIZE (1024*1024*1)
 /* rename RAnalFunctionBB.add() */
-R_API int r_anal_fcn_add_bb(RAnal *anal, RAnalFunction *fcn, ut64 addr, ut64 size, ut64 jump, ut64 fail, int type, RAnalDiff *diff) {
+R_API bool r_anal_fcn_add_bb(RAnal *anal, RAnalFunction *fcn, ut64 addr, ut64 size, ut64 jump, ut64 fail, int type, RAnalDiff *diff) {
 	RAnalBlock *bb = NULL, *bbi;
 	RListIter *iter;
 	bool mid = false;
 	st64 n;
+	if (size == 0) { // empty basic blocks allowed?
+		eprintf ("warning: empty basic block at 0x%08"PFMT64x" is not allowed. pending discussion.\n", addr);
```

Comment:
use a `r_warn_if_reached`

Reply:
this macro doesnt allows me to add any custom format string. its not even clear what will happen if that assert is reached when no logs are shown (will assert, crash, ignore?)

Reply:
mmm as the name implies, it just warns if the line is reached. But even in this case it would be probably better to use `r_warn_if_fail`. We can change it to accept a custom string.